mathesar_ui/src/pages/database-connection/DatabaseConnectionsList.svelte

```diff
@@ -0,0 +1,120 @@
+<script lang="ts">
+  import { databases } from '@mathesar/stores/databases';
+  import AppendBreadcrumb from '@mathesar/components/breadcrumb/AppendBreadcrumb.svelte';
+  import { iconAddNew, iconDatabase } from '@mathesar/icons';
+  import {
+    DATABASE_CONNECTION_ADD_URL,
+    DATABASE_CONNECTION_LIST_URL,
+  } from '@mathesar/routes/urls';
+  import type { Database } from '@mathesar/AppTypes';
+  import { States } from '@mathesar/api/utils/requestUtils';
+  import ErrorBox from '@mathesar/components/message-boxes/ErrorBox.svelte';
+  import EntityContainerWithFilterBar from '@mathesar/components/EntityContainerWithFilterBar.svelte';
+  import { AnchorButton, Icon } from '@mathesar/component-library';
+  import { labeledCount } from '@mathesar/utils/languageUtils';
+  import { isSuccessfullyConnectedDatabase } from '@mathesar/utils/preloadData';
+  import DatabaseConnectionSkeleton from './DatabaseConnectionSkeleton.svelte';
+  import { makeSimplePageTitle } from '../pageTitleUtils';
+  import DatabaseConnectionItem from './DatabaseConnectionItem.svelte';
+
+  let filterQuery = '';
+
+  $: isPreloaded = $databases.preload;
+  $: allDatabases = $databases.data;
+  $: databasesLoadStatus = $databases.state;
+  $: databasesLoadError = $databases.error;
+
+  function filterDatabase(_databases: Database[], query: string) {
+    function isMatch(database: Database, q: string) {
+      if (!isSuccessfullyConnectedDatabase(database)) {
+        return database.name.toLowerCase().includes(q);
+      }
+      return (
+        database.name.toLowerCase().includes(q) ||
+        database.db_name.toLowerCase().includes(q)
+      );
+    }
+    return _databases.filter((database) => {
+      if (query) {
+        const sanitizedQuery = query.trim().toLowerCase();
+        return isMatch(database, sanitizedQuery);
+      }
+      return true;
+    });
+  }
+
+  function handleClearFilterQuery() {
+    filterQuery = '';
+  }
+
+  $: filteredDatabases = filterDatabase(allDatabases ?? [], filterQuery);
+  $: filteredDatabasesCountText = filteredDatabases.length
+    ? `(${filteredDatabases.length})`
+    : '';
+</script>
+
+<svelte:head>
+  <title>{makeSimplePageTitle('Database Connections')}</title>
+</svelte:head>
+
+<AppendBreadcrumb
+  item={{
+    type: 'simple',
+    href: DATABASE_CONNECTION_LIST_URL,
+    label: 'Add Database Connection',
+    icon: iconDatabase,
+  }}
+/>
+
+<h1>Database Connections {filteredDatabasesCountText}</h1>
+
+<section class="connections-list-container">
+  {#if databasesLoadStatus === States.Loading && !isPreloaded}
+    <DatabaseConnectionSkeleton />
+  {:else if databasesLoadStatus === States.Done || isPreloaded}
```

Comment:
What is the reasoning behind checking `isPreloaded`?

I'd lean towards removing the `preload` flag, since it's not used for any particular operation at the moment as far as I can see. We've done a number of refactors and the `preload` flag is obsolete here.